frontend/src/containers/AddChart.tsx

```diff
@@ -320,12 +307,13 @@ function AddChart() {
     });
   }
 
-  if (state && state.json !== undefined) {
-    if (step !== 1) {
-      setStep(1);
-      onSelectStaticDataset();
+  useEffect(() => {
+    if (datasetType) {
+      reset({
+        datasetType,
+      });
     }
-  }
+  }, []);
```

Comment:
Shouldn't the `datasetType` variable be inside the dependency array?

Reply:
We only want to call reset after the initial render